docs/jwst/stpipe/config_asdf.rst

```diff
@@ -75,65 +89,61 @@ The first 5 lines, up to and including the "---" line, define the file as an
 ASDF file. The rest of the file is formatted as one would format YAML data.
 Being YAML, the last line, containing the three ``...`` is essential.
 
-Parameters
-++++++++++
+class and name
+++++++++++++++
+
+There are two required keys at the top level: ``class`` and ``parameters``.
+``parameters`` is discussed below.
 
-A step configuration requires one key, called ``parameters``, which
-contains all the parameters to pass onto the step.
+``class`` specifies the Python class to run.  It should be a
+fully-qualified Python path to the class.  Step classes can ship with
+``stpipe`` itself, they may be part of other Python packages, or they
+exist in freestanding modules alongside the configuration file.  For
+example, to use the ``SystemCall`` step included with ``stpipe``, set
+``class`` to ``stpipe.subprocess.SystemCall``.  To use a class called
+``Custom`` defined in a file ``mysteps.py`` in the same directory as
+the configuration file, set ``class`` to ``mysteps.Custom``.
 
-The only key required in the ``parameters`` block is ``class``. This defines
-which ``Step`` or ``Pipeline`` the parameters belong to. This key is used by
-``strun`` to determine which class to actually execute when given a
-configuration file.
+``name`` defines the name of the step.  This is distinct from the
+class of the step, since the same class of Step may be configured in
+different ways, and it is useful to be able to have a way of
+distinguishing between them.  For example, when Steps are combined
+into :ref:`stpipe-user-pipelines`, a Pipeline may use the same Step class
+multiple times, each with different configuration parameters.
 
-Another key that will often be seen is ``name``. This defines an alias to use
-for the class referenced by the ``class`` key. Pipelines use this alias to refer
-to their sub-steps.
+Parameters
+++++++++++
 
-All other keys are the parameters and their values to be used when the
-step/pipeline is run. The order of the parameters does not matter. Except for
-the ``class`` key, no other parameter needs to be specified. If not defined, the
-default, as defined in the code, will be used.
+``parameters`` contains all the parameters to pass onto the step. The order of
+the parameters does not matter. It is not necessary to specify all parameters
+either. If not defined, the default, as defined in the code or values from CRDS
+parameter references, will be used.
```

Comment:
As mentioned above, I think it's important to note that for a `pars-foo` reffile for an instrument with CRDS selectors, only those parameters that change from the code default based on instrument and CRDS selectors should be included.

This is important to have here, as already many of the `pars-foo` reffiles delivered do not do this and blindly list all the parameters, regardless of whether they are different based on the instrument or the CRDS selectors for that instrument pars file.  This is bad, since it means that the defaults in the code will never get applied, even if they get changed in the code during commissioning.  In order to get the new defaults (say for a global parameter that is not instrument-specific) the teams will have to deliver new `pars-foo` reffiles.

I see this is covered a bit below in "Completeness", but I don't think we can repeat this enough.

Reply:
Thanks for pointing out that there are already references submitted. Appropriate messaging to INS will be going out shortly.